src/Ng/CommandHelpers.cs

```diff
@@ -12,381 +13,170 @@
-        }
-
-        public static string GetRegistration(IDictionary<string, string> arguments, bool required = true)
-        {
-            string value;
-            TryGetArgument(arguments, Constants.Registration, out value, required);
-            return value;
-        }
-
-        public static string GetCatalogBaseAddress(IDictionary<string, string> arguments, bool required = true)
-        {
-            string value;
-            TryGetArgument(arguments, Constants.CatalogBaseAddress, out value, required);
-            return value;
-        }
-
-        public static string GetStorageBaseAddress(IDictionary<string, string> arguments, bool required = true)
-        {
-            string value;
-            TryGetArgument(arguments, Constants.StorageBaseAddress, out value, required);
-            return value;
-        }
-
-        public static string GetId(IDictionary<string, string> arguments, bool required = true)
-        {
-            string value;
-            TryGetArgument(arguments, Constants.Id, out value, required);
-            return value;
-        }
-
-        public static string GetVersion(IDictionary<string, string> arguments, bool required = true)
-        {
-            string value;
-            TryGetArgument(arguments, Constants.Version, out value, required);
-            return value;
-        }
-
-        public static bool GetUnlistShouldDelete(IDictionary<string, string> arguments, bool required = true)
-        {
-            return GetBool(arguments, Constants.UnlistShouldDelete, defaultValue: false, required: required);
-        }
-
-        public static bool GetVerbose(IDictionary<string, string> arguments, bool required = true)
-        {
-            return GetBool(arguments, Constants.Verbose, defaultValue: false, required: required);
-        }
-
-        public static bool GetBool(IDictionary<string, string> arguments, string argumentName, bool defaultValue, bool required = true)
-        {
-            bool result;
-            string argumentValue;
-            if (TryGetArgument(arguments, argumentName, out argumentValue, required))
-            {
-                result = bool.Parse(argumentValue);
-            }
-            else
-            {
-                result = defaultValue;
-            }
-
-            return result;
-        }
-
-        public static int GetInterval(IDictionary<string, string> arguments, int defaultInterval)
-        {
-            int interval = defaultInterval;
-            string intervalStr = string.Empty;
-            TryGetArgument(arguments, Constants.Interval, out intervalStr);
-            if (!int.TryParse(intervalStr, out interval))
-            {
-                interval = defaultInterval;
-            }
-
-            return interval;
-        }
-
-        public static DateTime GetStartDate(IDictionary<string, string> arguments)
+        public static StorageFactory CreateStorageFactory(IDictionary<string, string> arguments, bool verbose)
```

Comment:
>CreateStorageFactory [](start = 37, length = 20)

I hope all the methods bellow dont have logic changes. This is not covered by UTs, and it's going to be hard to find how slightly different behavior affects things in runtime. My recommendation: undo this.